lib/ontologies_linked_data/utils/notifier.rb

```diff
@@ -0,0 +1,125 @@
+module LinkedData
+  module Utils
+    class Notifier
+      def self.notify(options = {})
+        return unless LinkedData.settings.enable_notifications
+
+        headers = { 'Content-Type' => 'text/html' }
+        sender = options[:sender] || LinkedData.settings.email_sender
+        recipients = Array(options[:recipients]).uniq
+        raise ArgumentError, 'Recipient needs to be provided in options[:recipients]' if !recipients || recipients.empty?
+
+        # By default we override all recipients to avoid
+        # sending emails from testing environments.
+        # Set `email_disable_override` in production
+        # to send to the actual user.
+        unless LinkedData.settings.email_disable_override
+          headers['Overridden-Sender'] = recipients
+          recipients = LinkedData.settings.email_override
+        end
+
+        Pony.mail({
+                    to: recipients,
+                    from: sender,
+                    subject: options[:subject],
+                    body: options[:body],
+                    headers: headers,
+                    via: :smtp,
+                    enable_starttls_auto: LinkedData.settings.enable_starttls_auto,
+                    via_options: mail_options
+                  })
+      end
+
+      def self.notify_ontoportal_admins_grouped(subject, body)
+        notify_mails_grouped subject, body, ontoportal_admin_emails if LinkedData.settings.enable_administrative_notifications
+      end
+
+      def self.notify_subscribed_separately(subject, body, ontology, notification_type)
+        mails = subscribed_users_mails(ontology, notification_type)
+        notify_mails_separately subject, body, mails
+      end
+
+      def self.notify_administrators_grouped(subject, body, ontology)
+        notify_mails_grouped subject, body, ontology_admin_emails(ontology)
+      end
+
+      def self.notify_mails_separately(subject, body, mails)
+        mails.each do |mail|
+          notify_mails_grouped subject, body, mail
+        end
+      end
+
+      def self.notify_mails_grouped(subject, body, mail)
+        options = {
+          subject: subject,
+          body: body,
+          recipients: mail
+        }
+        notify(options)
+      end
+
+      def self.notify_ontoportal_admins(subject, body)
+        notify_mails_grouped subject, body, ontoportal_admin_emails if LinkedData.settings.enable_administrative_notifications
```

Comment:
no need of the `if LinkedData.settings.enable_administrative_notifications` as notify already test that

Reply:
`LinkedData.settings.enable_administrative_notifications` is not the same as `LinkedData.settings.enable_notifications`

`enable_notifications` enables all notifications for all users
`enable_administrative_notifications` enables notifications send to the admin team (or support team in your case) for things like new user creation event.  BioPortal team doesn't have the need to see those notifications.

Reply:
Ah ok sorry, miss reading the thing,
Good for me